design/specs/record-page.md

```diff
@@ -1,71 +1,137 @@
+
+    The button is a link (can be opened in a new tab) to create a new query within the Data Explorer. The link contains URL parameters which specify the id of this table and the id of this record. The Data Explorer then allow the user to build a query from scratch, but the choice of base table is limited only to tables which contain FKs to this table.
+
+    While building the query, the Data Explorer displays the Record Summary of the original record (where the user was before building the query). The user can click on that Record Summary to navigate back to that Record Page. The results of the query the user is building are filtered to show only the items related to that record. That filtering operation is not defined within the query the user is building, but rather performed as part of the `filter` params when fetching the results of the query. The UI which displays the specific Record Summary also allows the user to change the record (using the Record Selector) -- merely for the purpose of previewing their query from the perspective of a different record while building it.
+    
+    We will need to clarify what happens when the user saves the query. Is the query visible within the top level of the schema? Or is it hidden? Perhaps this is configurable.
+
+- Each **manually-generated widget** provides the following options:
+    - "Rename"
+    - "Edit"
+    - "Delete"
+
+- Each **auto-generated widget** provides the following options:
+    - "Hide"
+    - "Edit a copy of this query" -- This is a link similar to the "Add Query" button which directs the user to the Data Explorer, but it also passes the id of the foreign key used in the auto-generated widget. The Data Explorer then builds a query to match the behavior of the auto-generated widget and allows the user to save it as a new query. The auto-generated widget will remain unmodified and visible, leaving the user with two very similar widgets. At this point the user can choose to distinguish between the two by either hiding the auto-generated one or renaming the manually-generated one.
+
+- Above all widgets, a "Show Hidden Queries" dropdown button exists giving the user the capability to un-hide any auto-generated widgets widgets they have hidden.
+
+- Each widget is collapsible via a triangle icon to the left of its title. The collapsed/expanded state per-widget is persisted via local storage and will apply to all records within this table.
+    
+- The default ordering of widgets is not defined, but users *can* re-order all the widgets as as they so choose.
+
+    The mechanism for re-ordering the widgets is subject to further UX experimentation. Drag and drop would seem to be an obvious choice, though the height of each widget (when expanded) would make dragging cumbersome. It may be possible to collapse all widgets when the user initiates a drag, however the user's cursor might jump around on the page if we do this.
```

Comment:
As you point out, the height of the widgets will complicate drag-and-drop ordering. You may instead experiment with a menu that allows you to change the visibility and ordering of the widgets.
<img width="256" alt="image" src="https://user-images.githubusercontent.com/845767/181198705-26eaf0fe-4b25-4495-ae6b-deb15880f9ba.png">

Reply:
Cool idea! In 9a59ef10f964f3b04a1d9cd1f1d1335f5e16ab78 I've added the following text to the specs:

> Another idea is a drop down menu in which the user can re-order the widgets. This menu could potentially contain UI to show/hide the widgets too!